src/App.tsx

```diff
@@ -15,47 +15,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+interface ReorderOptions {
+  sortType: SortType,
+  isReversed: boolean,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods
+        .sort((firstGood, secondGood) => firstGood.localeCompare(secondGood));
+      break;
+    case SortType.LENGTH:
+      visibleGoods
+        .sort((firstGood, secondGood) => firstGood.length - secondGood.length);
+      break;
+    case SortType.NONE:
+      break;
+    default:
+      throw new Error('wrong sort type');
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const sortAlphabetically = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const reverse = () => {
+    setIsReversed(reversed => !reversed);
+  };
+
+  const reset = () => {
```

Comment:
check checklist 
```suggestion
  const handleReset = () => {
```
same for `reverse`